corehq/apps/reports/static/reports/v2/js/datagrid/data_models.js

```diff
@@ -22,6 +22,26 @@ hqDefine('reports/v2/js/datagrid/data_models', [
         self.isDataLoading = ko.observable(false);
         self.isLoadingError = ko.observable(false);
 
+        self.isDataLoading.subscribe(function (isLoading) {
+            if (!isLoading) return;
+
+            var $rows = $('#js-datagrid-rows'),
+                $loading = $('#js-datagrid-loading'),
+                position = $rows.position(),
+                marginTop = $rows.height()/2 - 50;
```

Comment:
Can you add a comment on what the `- 50` is accounting for?

Reply:
An overall comment on this block of code would also be helpful.

Reply:
done